catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/OrientationDialogTest.java

```diff
@@ -0,0 +1,128 @@
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.rules.DontGenerateDefaultProjectActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_CAST_GLOBALLY_ENABLED;
+
+@RunWith(AndroidJUnit4.class)
+public class OrientationDialogTest {
+
+	@Rule
+	public DontGenerateDefaultProjectActivityInstrumentationRule<MainMenuActivity> baseActivityTestRule = new
+			DontGenerateDefaultProjectActivityInstrumentationRule<>(MainMenuActivity.class, true, false);
+
+	private IdlingResource idlingResource;
+	private String chromeCast = SETTINGS_CAST_GLOBALLY_ENABLED;
+	private boolean initialChromeCastSetting;
+
+	@Before
+	public void setUp() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		SharedPreferences.Editor sharedPreferencesEditor = sharedPreferences.edit();
+
+		initialChromeCastSetting = sharedPreferences.getBoolean(chromeCast, false);
+
+		sharedPreferencesEditor.putBoolean(chromeCast, true);
+
+		sharedPreferencesEditor.commit();
+
+		baseActivityTestRule.launchActivity(null);
+
+		idlingResource = baseActivityTestRule.getActivity().getIdlingResource();
+		Espresso.registerIdlingResources(idlingResource);
+	}
+
+	@Test
+	//@category({Level.Smoke.class, Cat.AppUi.class, Cat.Gadgets.class})
+	@Category({Level.Smoke.class, Cat.AppUi.class, Cat.Gadgets.class})
+	public void newProject() {
+		onView(withId(R.id.main_menu_button_new))
+				.perform(click());
+
+		//check if dialog title is displayed
+		onView(withText(R.string.new_project_dialog_title))
+				.check(matches(isDisplayed()));
+
+		//enter new project name
+		onView(withId(R.id.project_name_edittext))
+				.perform(typeText("TestCastProject"));
+		onView(withText(R.string.ok))
+				.perform(click());
+
+		//check if orientation dialog is displayed
+		onView(withText(R.string.project_select_screen_title))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.cast)).perform(click());
+		onView(withText(R.string.ok))
+				.perform(click());
+
+		//check if user ends up in right activity either by checking activity itself:
```

Comment:
if this last check isnt necessary, please skip it.... (in this case also remove the click on "ok" above)